test/e2e/dryrun.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"sigs.k8s.io/kwok/pkg/utils/path"
+)
+
+func loadExpectedClusterDetails(filepath string) (string, error) {
+	data, err := os.ReadFile(filepath)
+	if err != nil {
+		return "", err
+	}
+	out := string(data)
+	out = strings.ReplaceAll(out, "\n", " ")
+	out = strings.Join(strings.Fields(out), " ")
```

Comment:
```suggestion
```

Reply:
@wzshiming if we remove this then the output and testdata will not match as in the output there are many newlines and extra spaces which should be removed before comparing

Reply:
The test data is written from the output so why wouldn't it match?

Reply:
Because I have removed the indentations and the actual output consists of many blank lines which I had removed from the testdata so testdata is not exactly same as the output.Do you want me the testdata to be the exact output that we get(it contains lots of blank lines and indentations) or we should keep it this way?